src/player.js

```diff
@@ -1288,6 +1299,30 @@ export default class Player extends FakeEventTarget {
     return false;
   }
 
+  /**
+   * update the ABR restriction config
+   * @function _applyABRRestriction
+   * @returns {void}
+   * @param {Object} config - new config which configure for checking if it relevant config has changed
+   * @private
+   */
+  _applyABRRestriction(config: Object): void {
+    if (Utils.Object.hasPropertyPath(config, 'abr.restrictions') && this._engine) {
+      const {restrictions} = this._config.abr;
+      const videoTracks = this._allTracks.filter(track => track instanceof VideoTrack);
+      const newVideoTracks = filterTracksByRestriction(videoTracks, restrictions);
+      if (newVideoTracks.length) {
+        const currentVideoTracks = this._availableTracks.filter(track => track instanceof VideoTrack);
+        this._engine.applyABRRestriction(restrictions);
+        if (JSON.stringify(currentVideoTracks) !== JSON.stringify(newVideoTracks)) {
+          this._availableTracks = this._allTracks.filter(track => !(track instanceof VideoTrack)).concat(newVideoTracks);
+          this.dispatchEvent(new FakeEvent(CustomEventType.TRACKS_CHANGED, {tracks: newVideoTracks}));
+        }
+      } else {
+        Player._logger.warn('nothing has changed values does not meet the restriction');
```

Comment:
```suggestion
        Player._logger.warn('Nothing has changed values do not meet the restriction');
```
I'm not sure `values` is clear term here